pygame_gui/elements/ui_form.py

```diff
@@ -0,0 +1,1144 @@
+            pos.x = -self.submit_button.rect.width - x_padding
+
+        self.submit_button.set_anchors(anchors)
+        self.submit_button.set_dimensions((self.submit_button.relative_rect.width, self.field_height))
+        self.submit_button.set_relative_position(pos)
+
+    def rebuild(self) -> None:
+        """
+        A complete rebuild of the drawable shape used by this element.
+
+        :return: None
+        """
+
+        super().rebuild()
+
+        root_rect = pygame.Rect(self.relative_rect.x + self.shadow_width + self.border_width,
+                                self.relative_rect.y + self.shadow_width + self.border_width,
+                                self.relative_rect.width - (2 * self.shadow_width) - (2 * self.border_width),
+                                self.relative_rect.height - (2 * self.shadow_width) - (2 * self.border_width))
+
+        if self._root_container.relative_rect != root_rect:
+            self._root_container.set_relative_position(root_rect.topleft)
+            self._root_container.set_dimensions(root_rect.size)
+
+        self.rebuild_parsed_questionnaire(self, self.parsed_questionnaire)
+        self.rebuild_submit_button()
+
+        theming_parameters = {"normal_bg": self.background_colour,
+                              "normal_border": self.border_colour,
+                              "normal_image": self.background_image,
+                              "border_width": self.border_width,
+                              "shadow_width": self.shadow_width,
+                              "shape_corner_radius": self.shape_corner_radius}
+
+        if self.shape == "rectangle":
+            self.drawable_shape = RectDrawableShape(self.rect, theming_parameters,
+                                                    ["normal"], self.ui_manager)
+        elif self.shape == "rounded_rectangle":
+            self.drawable_shape = RoundedRectangleShape(self.rect, theming_parameters,
+                                                        ["normal"], self.ui_manager)
+
+        self.on_fresh_drawable_shape_ready()
+
+    def process_event(self, event: pygame.event.Event) -> bool:
+        """
+        Handles the submission of the form
+
+        :param event: The event to process
+        :return: Whether the event was consumed
+        """
+
+        consumed = False
+        if event.type == UI_BUTTON_PRESSED and event.mouse_button == pygame.BUTTON_LEFT:
+            if event.ui_element == self.submit_button:
+                form_values = self.get_current_values()
+
+                # old event - to be removed in 0.8.0
+                event_data = {"user_type": OldType(UI_FORM_SUBMITTED),
+                              "form_values": form_values,
+                              "ui_element": self,
+                              "ui_object_id": self.most_specific_combined_id}
+                pygame.event.post(pygame.event.Event(pygame.USEREVENT, event_data))
```

Comment:
```suggestion
```
We don't need to add the old event type to new code, it's just there in older code for backwards compatibility.